src/main/java/com/synopsys/integration/alert/provider/blackduck/descriptor/BlackDuckProviderUIConfig.java

```diff
@@ -54,12 +54,14 @@ public BlackDuckProviderUIConfig() {
     }
 
     private Collection<String> validateAPIToken(final FieldValueModel fieldToValidate, final FieldModel fieldModel) {
-        final String apiKey = fieldToValidate.getValue().orElse("");
-        if (StringUtils.isBlank(apiKey)) {
-            return List.of(ConfigField.REQUIRED_FIELD_MISSING);
-        } else {
-            if (apiKey.length() < 64 || apiKey.length() > 256) {
-                return List.of("Invalid Black Duck API Token.");
+        if (fieldToValidate.hasValues() || !fieldToValidate.isSet()) {
```

Comment:
Why do we need this if? and is it doing what you expect? the condition seems very confusing

Reply:
if there are values we want to validate if the list is empty and isSet is false we want to validate.  When isSet=true because it is saved the values list is empty which is ok.